rewriter/RequiresAncestorKernel.cc

```diff
@@ -25,6 +25,11 @@ void RequiresAncestorKernel::run(core::MutableContext ctx, ast::ClassDef *klass)
         return;
     }
 
+    auto id = ast::cast_tree<ast::UnresolvedConstantLit>(klass->name);
+    if (id && id->cnst == core::Names::Constants::Kernel()) {
```

Comment:
This should also check for `id` having a null or root `scope`?  Because you want:

```
module Mine::Kernel
end
```

to get the `requires_ancestor` call added?